### PR TITLE
Implement UserAgent fork using ua-parser-js

### PR DESCRIPTION
**Pull request by zpao**
The goal here is to approximate our internal implementation, which is
all done server-side. This will not be perfect but we'll patch things up
as we need to.

We opted for ua-parser-js since it matches up pretty well with the handful of UAs we threw at it, has a fairly small footprint, is popular seems to be well supported.

cc @salier

### CODE REVIEW

**Comment by zpao** [comment]
Another thing we might be able to do is match up more closely with our code so `UserAgent` stays in line with our internal code and we just have a `UserAgentData` module that exports the data (which is how it works internally). I was trying to avoid actually parsing at `require` time though, but it might not actually matter…

**Comment by zpao** [comment]
Alright, after working on this some more, we decided to just keep our internal UserAgent module the same and instead implement UserAgentData. The downside of this is that it will be calculated at require time, but that shouldn't be too bad.

**Comment by spicyj** [comment]
crazytown!

File: src/__forks__/UserAgentData.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * Usage note:
+ * This module makes a best effort to export the same data we would internally.
+ * At Facebook we use a server-generated module that does the parsing and
+ * exports the data for the client to use. We can't rely on a server-side
+ * implementation in open source so instead we make use of an open source
+ * library to do the heavy lifting and then make some adjustments as necessary.
+ * It's likely there will be some differences. Some we can smooth over.
+ * Others should are going to be harder.
```
Comment (wallclockbuilder) [inline src/__forks__/UserAgentData.js]: Others ~~should~~ are going to be harder.

Comment (zpao) [inline src/__forks__/UserAgentData.js]: Thanks. Fixed with 2ff5001b4d461f9c84ed8bbd88cf5f706b7f46aa

**Pull request merged**